### ISSUE TITLE
Add C callback in r_bp

### ISSUE THREAD

**Issue by radare** [radareorg/radare2#946]
Used for tracing, breakpointing etc, and running code from inside the api. this will make the debugger loop async and we will require to add other callbacks to handle other exceptions instead of checking right after the `waitpid()` thing

**Comment by XVilka** [comment]
@jduck your thoughts?

**Comment by jduck** [comment]
I'll take this into consideration surely.

**Comment by jduck** [comment]
After thinking some more about the issue, I think it's quite doable to have a callback function that gets called whenever a breakpoint is hit. That said, such a thing is only useful when using the r_debug API directly via C or bindings, right? In all likelihood the code would be executed right before "cmd.bp" is executed, so it wouldn't exactly be async. If something bad happens during processing of this breakpoint all bets would be off. For example, the user could kill the child or detach the inferior. If they do that, things are going to get really ugly when we attempt to proceed processing the breakpoint handling...

**Comment by radare** [comment]
yep. that callback will be only usable when using the api, but my idea was to use it interanlly because of those hairy corners

**Comment by jduck** [comment]
Luckily the handling of when breakpoints are hit is much simpler than continuing afterward. hehe.

**Comment by radare** [comment]
ping

**Comment by radare** [comment]
@jduck ping?

**Comment by jduck** [comment]
Sorry, no time lately :-/

**Comment by radare** [comment]
cc @leberus
